libeos-update-server/server.c

```diff
@@ -255,15 +255,14 @@ update_pending_requests (EusServer *self,
   GObject *obj = G_OBJECT (self);
 
   g_assert (delta >= 0 || self->pending_requests >= (guint) -delta);
-  g_assert (delta <= 0 || self->pending_requests <= G_MAXUINT - delta);
+  g_assert (delta <= 0 || self->pending_requests <= G_MAXUINT - (guint) delta);
 
   if (delta == 0)
     return;
 
-  g_debug ("%s: Updating from %u to %u", G_STRFUNC, self->pending_requests,
-           self->pending_requests + delta);
+  g_debug ("%s: Updating from %u by %d", G_STRFUNC, self->pending_requests, delta);
 
-  self->pending_requests += delta;
+  self->pending_requests += (guint) delta;
```

Comment:
It looks like `delta` can be negative here here (based on the `g_assert` above) but it is cast to a `guint`

Reply:
I *think* this will do what we want, since it’s just making the cast which was previously implicit, explicit. In two’s-complement terms, the addition will deliberately wrap around and give the right answer. However, I will double-check.

Reply:
Yes, this does do what we want, although I agree that it’s quite counterintuitive. I’ll add a comment.

Reply:
Done.